### DISCUSSION TITLE
Is there a way to resume pipeline execution?

### DISCUSSION THREAD

**Discussion by wiesehahn** [r-lidar/lasR#41] [General]
Is there a good/ recommend way to resume a pipeline execution for unprocessed data when working with VPC?

Background is that I ran a processing pipeline on a folder with large amount of tiles (via VPC) and execution was aborted at some point (not due to lasR). 

Now I have a bunch of tiles which were processed and written to disk but still many unprocessed data tiles. 

With lidR I once did something like the following which made use of the `processed` attribute in the catalog, which we dont have in VPC.

```r
# dont run for tiles already processed
processed_images <- list.files(storage_path, pattern = ".tif")
processed_images <- substr(processed_images, 1, 14)

ctg@data <- ctg@data %>%
  mutate(processed = case_when(paste0(as.integer(Min.X), "_", as.integer(Min.Y)) %in% processed_images ~ FALSE,
                               sf::st_intersects(ctg@data, aoi, sparse = FALSE) ~ TRUE,
                               TRUE ~ FALSE))
```

I could imagine something like a temporary file which is written and updated with information about finished jobs while processing. This could be deleted at end of pipeline execution and remaining on disk if R crashes or similar. Not sure if this makes sense though.

**Comment by Jean-Romain** [comment]
lidR has a mechanism to resume

```r
opt_restart(ctg) <- 50 # restart at chunk 50
```

But in lasR there is nothing yet. And it will be much harder to implement. Yet, this is something I'll consider for later. Added in my todo list https://github.com/orgs/r-lidar/projects/5/views/1

**Reply by wiesehahn** [reply]
Not sure if the link to the todo list was meant be accessible. It is not.

**Reply by Jean-Romain** [reply]
The link is now public
